plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -412,6 +415,8 @@ export async function createRouter(
       if (!task) {
         throw new NotFoundError(`Task with id ${taskId} does not exist`);
       }
+      // Do not disclose secrets
+      delete task.secrets;
```

Comment:
it would be great to get a test ensuring that no secret reach the caller, primary to ensure that we don't remove this logic by misstake in the future..

Reply:
Good suggestion. Added a test case for this.

Reply:
Kinda makes me wonder if it'll be reintroduced elsewhere (e.g. in a list call or so) later. Could also have added an `options: { includeSecrets?: boolean }`, but not gonna block on that

Reply:
Would it make sense to refactor out further into its own database table for this specific reason, e.g. explicit code required to fetch credentials to avoid mistakes leading to security vulnerabilities?

Reply:
maybe as a followup - but we may get far by just gating with an opt-in param of some sort.